examples/Misc/Program.cs

```diff
@@ -34,7 +34,7 @@ static void ListGroups(string brokerList)
 
             using (var producer = new Producer(config))
             {
-                var groups = producer.ListGroups(TimeSpan.FromSeconds(10));
+                var groups = producer.ClientInstance.ListGroups(TimeSpan.FromSeconds(10));
```

Comment:
I really dont like this, it looks like internal implementation details bleeding through to the public API.
From a user's perspective I'd much rather have those methods on my Producer instance, as is today, but using a base-class to avoid the repetetiveness.
This would also make it easier to make integrate with the library since you could make a method that takes a Client instance, rather than one for Producer or one for Consumer.